### PR TITLE
Allow newer dependencies

### LINKED ISSUE #100: network-3.2 not allowed
network-3.2 was released on hackage, but this library doesn't allow it. See:

* https://github.com/commercialhaskell/stackage/issues/7381

Note that Hackage maintainers can now [receive notifications when dependencies go out of bound](https://discourse.haskell.org/t/hackage-dep-bound-notifications-and-uploader-endorsements-for-new-users/8342?u=janus).

### PR DESCRIPTION
**Pull request by ysangkok**
Tested using

    cabal test -w ghc-9.10.1 -c 'network>=3.2' -c 'lens>=5.3'

Fixes #100

### CODE REVIEW

**Comment by ysangkok** [comment]
@maoe  I pushed a commit to update the CI config, it should work now.

**Comment by maoe** [comment]
Apparently the last commit removed some influxdb related settings by accident.

**Comment by ysangkok** [comment]
@maoe I have put it back

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -66,69 +71,47 @@ jobs:
...
+          chmod a+x "$HOME/.ghcup/bin/ghcup"
+          "$HOME/.ghcup/bin/ghcup" install ghc "$HCVER" || (cat "$HOME"/.ghcup/logs/*.* && false)
+          "$HOME/.ghcup/bin/ghcup" install cabal 3.12.1.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
         env:
           HCKIND: ${{ matrix.compilerKind }}
           HCNAME: ${{ matrix.compiler }}
           HCVER: ${{ matrix.compilerVersion }}
-      - name: influxdb forward
-        run: |
-          socat TCP-LISTEN:8086,fork TCP:influxdb:8086 &
```
Comment (maoe) [inline .github/workflows/haskell-ci.yml:108]: This block as well

**Review by maoe — CHANGES_REQUESTED** [review]

File: .github/workflows/haskell-ci.yml (hunk 1 above)
Comment (ysangkok) [reply .github/workflows/haskell-ci.yml]: @maoe, I have also added the installation of socat, which is necessary for the forwarding.

**Comment by ysangkok** [comment]
@maoe Sorry this took so many iterations to fix, but I have now verified in my fork that this last version in the latest commit actually works: https://github.com/ysangkok/influxdb-haskell/actions/runs/9886015050

**Comment by maoe** [comment]
LGTM. Thanks!

**Comment by maoe** [comment]
Uploaded https://hackage.haskell.org/package/influxdb-1.9.3.2

**Pull request merged**